src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/SimpleHttpClientFactory.cs

```diff
@@ -16,7 +18,7 @@ namespace Microsoft.Identity.Client.PlatformsCommon.Shared
     /// .NET should use the IHttpClientFactory, but MSAL cannot take a dependency on it.
     /// .NET should use SocketHandler, but UseDefaultCredentials doesn't work with it 
     /// </remarks>
-    internal class SimpleHttpClientFactory : IMsalMtlsHttpClientFactory
+    internal class SimpleHttpClientFactory : IMsalMtlsHttpClientFactory, IMsalSFHttpClientFactory
```

Comment:
We should avoid having a non-liner inheritance tree around these interfaces, they will be hard to manage. Can we have 

IMsalMtlsHttpClientFactory inherit from IMsalSFHttpClientFactory instead?

Reply:
The two interfaces seem different to me - `IMsalMtlsHttpClientFactory` provides the full certificate for setting it as the client certificate and `IMsalSFHttpClientFactory` provides just a handler for the purpose of setting the cert validation callback.

Reply:
agreed, if one interface were truly a superset of the other, inheritance could make sense. it is not in this case.